### PR TITLE
Changed HMC definitions to use Ansible compatible inventory & vault format

### PR DESCRIPTION
**Pull request by andy-maier**
See commit message.

TODOs:
* Support omitting userid/password for mocked HMCs. - DONE
* Verify inventory & vault files with zhmc ansible collection. - DONE
* Improve test coverage for testutils. - DONE

### CODE REVIEW

File: examples/example_hmc_vault.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# format and define specific variables for HMC authentication.
+#
+# Brief description of the file format:
+#
+#   hmc_auth:
+#     <hmc_name>:  # DNS hostname, IP address, or nickname of HMC
+#       hmc_userid: <hmc_userid>
+#       hmc_password: <hmc_password>
+#       hmc_verify: <hmc_verify>
+#       hmc_ca_certs: <hmc_ca_certs>
```
Comment (andy-maier) [inline examples/example_hmc_vault.yaml:19]: The auth parameters need to be named so they are compatible with the hmc_auth parameter of the zhmc Ansible modules, i.e. `userid`, `password`, `verify`, `ca_certs`.

**Review by vkathir82 — CHANGES_REQUESTED** [review]
For ansible modules to work changes required in example_hmc_inventory.yaml format

File: examples/example_hmc_vault.yaml (hunk 1 above)
Comment (andy-maier) [reply examples/example_hmc_vault.yaml]: done

File: examples/example_hmc_inventory.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+#
+# Notes for this example file:
+# * To use this example file, copy it to `~/.zhmc_inventory.yaml` which is the
+#   default path name used e.g. by the example scripts in the examples
+#   directory.
+
+all:
+  hosts:
+
+  HMC1:
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply examples/example_hmc_inventory.yaml]: done

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 84ef083d23ff0a23bde62c73a5a5cd183f25cd67
  * Your commit message body contains a line that is longer than 80 characters

Guidelines are available at https://github.com/zhmcclient/python-zhmcclient/blob/master/CONTRIBUTING.rst

---
This message was auto-generated by https://gitcop.com

**Pull request merged**